clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -6191,8 +6181,28 @@ void CodeGenModule::EmitGlobalVarDefinition(const VarDecl *D,
         AddGlobalSYCLIRAttributes(GV, RD);
       // If VarDecl has a type decorated with SYCL device_global attribute 
       // emit IR attribute 'sycl-unique-id'.
-      if (RD->hasAttr<SYCLDeviceGlobalAttr>())
+      if (RD->hasAttr<SYCLDeviceGlobalAttr>()) {
         addSYCLUniqueID(GV, D, Context);
+        if (Linkage == llvm::GlobalValue::InternalLinkage) {
```

Comment:
Hmm....This seems ok to me but I am not sure if there are repercussions I am not thinking of . @AaronBallman can you weigh in here?